src/scripts/main.js

```diff
@@ -153,7 +153,8 @@
     // Extract map mice from a map
     function getMapMice(data, uncaught_only) {
         const mice = {};
-        $.each(data.treasure_map.goals.mouse, (key, mouse) => {
+        const map_things = data.treasure_map.is_scavenger_hunt ? 'item' : 'mouse';
+        $.each(data.treasure_map.goals[map_things], (key, mouse) => {
```

Comment:
We're just doing a reduce here:
```js
const key = data.treasure_map.is_scavenger_hunt ? 'item' : 'mouse';
const contents = Object.values(data.treasure_map[key])
  .reduce((dict, val) => {
    dict[val.unique_id] = val.name;
    return dict;
  }, {});
```
Don't forget that the "uncaught only" logic needs to use this key, too.

Reply:
if it's the same logic, I'd like to keep my version as it seems more beginner friendly. Or is there something i'm missing?

Reply:
The important callout is that the `uncaught_only == true` branch appears to assume the treasure map has mice data and seems like it will not work for scavenger maps (`hunter.completed_goal_ids.mouse`).

I prefer using JS builtins to jQuery whenever possible, so I provided a native alternative. I was also abstracting the variable names, since the dictionary may now contain items and not always mice.